ladybug/color.py

```diff
@@ -322,132 +313,113 @@ class ColorRange(object):
         >> <R:234, G:38, B:0>
 
         ##
-        colorRange = ColorRange(chart_type = 1)
+        colorRange = ColorRange(continuous_colors=False)
         colorRange.domain = [100, 2000]
         colorRange.colors = [Color(75, 107, 169), Color(245, 239, 103),
             Color(234, 38, 0)]
         colorRange.color(300)
         >> <R:245, G:239, B:103>
-
-        ##
-        colorRange = ColorRange(chart_type = 2)
```

Comment:
@chriswmackey before I start to review am I understand this correctly that you are removing ordinal charts? and if yes why are we doing that?

Reply:
@mostaphaRoudsari ,

I removed this only from the ColorRange object and I put the ability to create ordinal legends on the LegendPar and Legend classes through a dictionary that maps values to text categories:
https://github.com/ladybug-tools/ladybug/pull/141/files#diff-600acdaadb3713444a5c0a272f102834R285

My reasons for this are several-fold:

1) I know that you originally were thinking that things like data collections would have text data in them but, at this point, all data collections and data types are essentially numerical, though some data types [carry dictionaries that map values to text categories as you see here](https://github.com/ladybug-tools/ladybug/blob/master/ladybug/datatype/thermalcondition.py#L46-L47).  These dictionaries are essentially what will go into the LegendParameter ordinal_dictionary by default when no one has been specified.

2) With the way I have implemented the ordinal_dictionary now, users are forced to relate the text categories to numbers, making it possible to tell what the order of the text categories should be.  This was an issue before since it was difficult to tell which categories were "smaller" or "larger" than others. Having a dictionary like this also keeps things flexible if one decides that they don't want to display the full range of text categories that is available in the dictionary (ie. you only care about a PMV between -1 and +1 and not -3 and +3).  If you wanted to do this previously, you would have to edit the domain of the color range AND the values that you sent to the color range.  Now, you just change the min and/or max of the LegendParameters and it should all work.

3) If all that you really need is an object that maps text categories to colors, you don't need a whole ColorRange object capable of calculating unique colors for numerical values. Just use a dictionary.  It's just not worth it to add all of these extra `if` statements to the ColorRange object just to make it capable of doing something that a simple dictionary would do better.

Reply:
Two questions:

1. How does this approach will work for non-datatype data?
2. How would we map range of data to ordinal values? For instance < 300 ('low'), between 300-2000 ('desired') and > 2000 as ('too much')? Can we handle that?

Reply:
PS: Also see d3js scale categories:

https://github.com/d3/d3-scale#api-reference

Reply:
@mostaphaRoudsari ,

To answer your questions:

1. I am not sure I understand the question since the legend and color range exist independently of the datatype module.  They are built to take a set of any numerical data and make colors/ legends with it. So, if you want your legend to display with text values, and you don't have an ordinal dictionary from a DataType, you just make your own ordinal dictionary and assign it to the `ordinal_dictionary` of the LegendParameters.  We can have a component or a special way of dealing with Grasshopper inputs to help people make these dictionaries.

2. I am not sure if you are asking how to make a legend that displays with these text categories or if you are asking how can we produce colors that fall into 3 categories from a continuous data set.  So I will answer both. The first one I described above and in which case you'd make a legend with 3 segments at min/max values of 300 and 2000 and an ordinal_dictionary like so `{300: 'low', 1150: 'desired', 2000: 'too much'}`.  You can also make it with 5 segments and you will just get blank values for two of the segments in between the categories. For the second, you will see the [color range still has an option for continuous or discrete colors](https://github.com/ladybug-tools/ladybug/pull/141/files#diff-604a419b183a1c82099ebc5480cb0241R324).  The default is the continuous version we are used to but, if you only want to produce only 3 colors from the ColorRange, you just input 3 colors to the ColorRange and set the `continuous ` to False.

Reply:
Thanks.

1. Makes sense and it is a good approach.

2. I don't really follow all the details here. For instance I'm not sure why did you put 1150 there as a threshold but if I understand correctly it is possible to do so. Since this will be a common case I think it is helpful to add a sample code in tests to explain the workflow and also ensure it does work as expected.

Reply:
2. I put 1150 there because it is the midpoint between 300 and 2000 and it is a linear scale.  I will add some code in the tests that demonstrate how to do this and I will adapt the most useful examples for the `Usage:` part of the docstring.  

The point I made above does bring up one limitation of what I have implemented now, which is that the CorlorRanges is assumed to be a linear scale (it is not possible to do exponential or logarithmic scales now).  For now, I can't really think of cases where we really want an exponential or logarithmic scale but, if we want to add this in the future, the presumption would be that we add a special type of color range object that accepts the exponent and offset of a power scale (or vice versa for the log scale).  We can then add an option to the LegendParameters that changes the base color scale to one of these objects.

Reply:
>  For now, I can't really think of cases where we really want an exponential or logarithmic scale but, if we want to add this in the future, the presumption would be that we add a special type of color range object that accepts the exponent and offset of a power scale (or vice versa for the log scale). We can then add an option to the LegendParameters that changes the base color scale to one of these objects.

Or we can just keep ordinal option which gives you full freedom for such cases. There are several cases that the values are not linear and having an ordinal legend makes it quite easy to handle such cases. That is why I put it in legend in the first place.

Reply:
I think you are confusing the generation of colors for a set of values with the display of the legend.  You can create a legend that displays exponential values with the ordinal_dictionary.  I was talking about the generation of colors according to a power or log scale (something that was not there in the previous ColorRange and is not there now).

The ordinal legend that you originally had would not let you set up an power or log color scale because you need to let the user specify the exponent and offset if you are actually going to enable this.  Your original scale (and the one there now) lets you approximate a power or log scale by plugging in a list of sequentially exponential values for the domain but the interpolation between the values of each domain member is still linear. So it's not a true exponential or log scale.

Maybe I should just hold off on giving more of a description until you have the time to further review the code as it sounds like I create more confusion than clarity when I explain things.

Reply:
... or perhaps what will make it clearest is if I add some `Usage` cases.  I'll do that shortly.

Reply:
I have added a couple of resources (noted in the reply below) that should make things much clearer.

Reply:
Thank you for posting the sample code. The second sample shows that it is possible to do what I was looking for. I still think the set-up for ordinal dict is not clean. You have to come up with 1150 which is not really a number of interest. I get the logic of how you came up with this number but from user point of view having an explicit domain of 300 to 2000 is easier to understand and what I expect to put in if that is the range. In any case, since it works we can keep it as is.

I leave this part of the code a reference.

```
        data = [100, 300, 500, 1000, 2000, 3000]
        ordinal_dict = {300: 'low', 1150: 'desired', 2000: 'too much'}
        legend = Legend(data, LegendParameters(min=300, max=2000, number_of_segments=3,
                                               ordinal_dictionary=ordinal_dict))
        print(legend.segment_colors)
        legend.legend_parameters.continuous_colors = False  # get data in only 3 colors
        print(legend.value_colors)  # data colors align with number_of_segments
        >> ((R:75, G:107, B:169), (R:115, G:147, B:202), (R:115, G:147, B:202),
            (R:115, G:147, B:202), (R:115, G:147, B:202), (R:234, G:38, B:0))
```